tests/rptest/tests/data_migrations_api_test.py

```diff
@@ -214,3 +255,283 @@ def test_creating_and_listing_migrations(self):
         # todo: fix rp_storage_tool to use overridden topic names
         self.redpanda.si_settings.set_expected_damage(
             {"ntr_no_topic_manifest", "missing_segments"})
+
+    @property
+    def msg_size(self):
+        return 128
+
+    @property
+    def msg_count(self):
+        return int(20 * self.producer_throughput / self.msg_size)
+
+    @property
+    def producer_throughput(self):
+        return 1024 if self.debug_mode else 1024 * 1024
+
+    def start_producer(self, topic):
+        self.logger.info(
+            f"starting kgo-verifier producer with {self.msg_count} messages of size {self.msg_size} and throughput: {self.producer_throughput} bps"
+        )
+        producer = KgoVerifierProducer(
+            self.test_context,
+            self.redpanda,
+            topic,
+            self.msg_size,
+            100000000,  #we do not want to limit number of messages
+            rate_limit_bps=self.producer_throughput)
+
+        producer.start(clean=False)
+
+        wait_until(lambda: producer.produce_status.acked > 10,
+                   timeout_sec=120,
+                   backoff_sec=1)
+        return producer
+
+    def start_consumer(self, topic):
+        consumer = KgoVerifierConsumerGroupConsumer(self.test_context,
+                                                    self.redpanda,
+                                                    topic,
+                                                    self.msg_size,
+                                                    readers=3,
+                                                    group_name="test-group")
+
+        consumer.start(clean=False)
+        return consumer
+
+    def _do_validate_topic_operation(self, topic: str, op_name: str,
+                                     expected_to_pass: bool,
+                                     operation: Callable[[str], typing.Any]):
+        self.logger.info(
+            f"Validating execution of {op_name} against topic: {topic}")
+        success = True
+        try:
+            operation(topic)
+        except Exception as e:
+            self.logger.info(
+                f"Operation {op_name} executed against topic: {topic} failed - {e}"
+            )
+            success = False
+
+        assert expected_to_pass == success, f"Operation {op_name} outcome is not \
+            expected. Expected to pass: {expected_to_pass}, succeeded: {success}"
+
+    def validate_topic_access(
+        self,
+        topic: str,
+        metadata_locked: bool,
+        read_blocked: bool,
+        produce_blocked: bool,
+        test_add_partitions: bool = True,
```

Comment:
it's always the same as `metadata_locked`

Reply:
i guess it is not for created outbound migration

Reply:
No I cannot find a call to this function where `test_add_partitions` and `metadata_locked` have different values. If you can could you show me please?

Reply:
line `423` in `tests/rptest/tests/data_migrations_api_test.py`

Reply:
it has `metadata_locked=True`  and `test_add_partitions` is True by default